config/config.go

```diff
@@ -111,6 +117,140 @@ func GetNodeConfig(nodeConf *NodeConfig, cmConf *cmcfg.Config) {
 	}
 }
 
+// UpdateNodeConfigWithFlags updates NodeConfig with values from command line flags.
+func UpdateNodeConfigWithFlags(nodeConf *NodeConfig, flags *pflag.FlagSet) (err error) {
+	updateBoolFlag := func(flagName string, updateFunc func(bool)) {
+		if flag := flags.Lookup(flagName); flag.Changed {
+			updateFunc(flag.Value.String() == "true")
+		}
+	}
```

Comment:
KISS: use pointers instead of lambdas. This simplifies the code a lot, especially at calling side.
```suggestion
	updateBoolFlag := func(flagName string, target *bool) {
		if flag := flags.Lookup(flagName); flag.Changed {
			*target = flag.Value.String() == "true"
		}
	}
```